mockredis/client.py

```diff
@@ -75,6 +75,30 @@ def pipeline(self, transaction=True, shard_hint=None):
         """Emulate a redis-python pipeline."""
         return MockRedisPipeline(self, transaction, shard_hint)
 
+    def transaction(self, func, *watches, **kwargs):
```

Comment:
looks good to me but one concern I have is that this method is only applicable for StrictRedis class so can you add logic to raise AttributeError when it is not strict?